### PR TITLE
[GEOS-10533] Review startup warning, info and config logging

### PR DESCRIPTION
**Pull request by jodygarnett**
[GEOS-10533](https://osgeo-org.atlassian.net/browse/GEOS-10533)

Also cut back on shutdown messages

Message on the management of threads and calling of destroy() are intended for developers and have been reduced to fine logging level.

See also https://github.com/GeoWebCache/geowebcache/pull/1073

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public GeoserverXMLResourceProvider(
             throws ConfigurationException {
         this.configFileName = configFileName;
         this.configDirectory = inferConfigDirectory(resourceStore, providedConfigDirectory);
-        LOGGER.info(
+        LOGGER.debug(
```
Comment (bradh) [inline src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java:52]:
```suggestion
        LOGGER.config(
```

File: src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java [hunk 2]
```diff
@@ -154,7 +154,7 @@ private Resource findOrCreateConfFile() throws IOException {
         Resource xmlFile = findConfigFile();
 
         if (Resources.exists(xmlFile)) {
-            LOGGER.info("Found configuration file in " + configDirectory.path());
+            LOGGER.debug("Found configuration file in '" + configDirectory.path() + "'");
```
Comment (bradh) [inline src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java:157]:
```suggestion
            LOGGER.config("Found configuration file in '" + configDirectory.path() + "'");
```

**Review by bradh — APPROVED** [review]
Couple of suggestions for consistency in level. Otherwise OK.

File: src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java (hunk 1 above)
Comment (jodygarnett) [reply src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java:52]: Sadly I tried that, but this is a query method which is called multiple times. So the log ends up with the same message multiple times.

File: src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java (hunk 2 above)
Comment (jodygarnett) [reply src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java:157]: This one may be appropriate, I will test and report back.

Comment (jodygarnett) [reply src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java]: It was actually the other way around, this oone has to stay at debug.

I found that this was using the log4j 2 api directly and updated it to follow our logging standards.

File: src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java (hunk 1 above)
Comment (jodygarnett) [reply src/gwc/src/main/java/org/geoserver/gwc/config/GeoserverXMLResourceProvider.java]: Actually this was is okay, adds a little noise in the logs but does not repeat; changed to config logging level.

**Pull request merged**